context.go

```diff
@@ -1549,6 +1549,9 @@ func (c *Context) resolveDependencies(ctx context.Context, config interface{}) (
 		if len(errs) > 0 {
 			return
 		}
+		// if any presingletons visited all modules, the cache would be populated by only modules that exist pre-mutators.
+		// we clear the cache so that future calls to visit all modules will be accurate
+		c.clearSortedModuleGroupsCache()
```

Comment:
Whenever a new module is added we increment the atomic c.depsModified variable, can we make sortedModuleGroups() automatically clear the cache if that variable is set, or add a new one if the interactions with the existing users of that variable are a problem?

Reply:
Good idea. Done.

Reply:
Further updated to address your comments about depsModified being reset.